graphene_django/fields.py

```diff
@@ -90,6 +90,10 @@ def connection_resolver(cls, resolver, connection, default_manager, max_limit,
         if isinstance(iterable, QuerySet):
             if iterable is not default_manager:
                 default_queryset = maybe_queryset(default_manager)
+                if default_queryset.query.distinct and not iterable.query.distinct:
```

Comment:
How about moving this logic into `merge_querysets`? Then any of the subclasses that use that will get this fix for free.

Reply:
Good idea, did that